src/NUnitFramework/framework/Internal/Builders/NUnitTestCaseBuilder.cs

```diff
@@ -228,6 +228,9 @@ private static bool CheckTestMethodSignature(TestMethod testMethod, TestCasePara
                 parameters = testMethod.Method.GetParameters();
             }
 
+            if (parms != null && parms.TestName != null && parms.TestName.Trim() == "")
+                return MarkAsNotRunnable(testMethod, "Test has no name");
```

Comment:
```suggestion
                 return MarkAsNotRunnable(testMethod, "Test name cannot be all white-space or empty.");
```

I'm not sure on "Test has no name" - how about something more explicit like this instead? 🙂